--- /dev/null	2024-02-14 20:31:11.878000303 +1100
+++ cfg-if-1.0.0/meson.build	2024-02-14 20:31:51.693790551 +1100
@@ -0,0 +1,24 @@
+project('generic-array-rs', 'rust',
+  version : '0.14.7',
+  meson_version : '>= 1.3.1',
+  default_options : ['buildtype=debugoptimized',
+                     'rust_std=2015'
+                    ]
+)
+
+rustc = meson.get_compiler('rust')
+
+typenum_dep = dependency(
+  'typenum-rs',
+  version : ['>= 1.12', '< 2']
+)
+lib = static_library(
+  'generic_array',
+  'src/lib.rs',
+  dependencies: [typenum_dep],
+  rust_dependency_map : {}
+)
+
+dep = declare_dependency(link_with : lib)
+meson.override_dependency('generic-array-rs', dep)
+
